File: CS/CS.NET-VB.NET/Visual Studio/IHttpHandler/IHttpHandler/HttpHandlerVB/CustomHandler.ashx.vb
Imports System.Web
Imports System.Web.Services


Public Class CustomHandler
    Implements System.Web.IHttpHandler

    Sub ProcessRequest(ByVal context As HttpContext) Implements IHttpHandler.ProcessRequest
        If context.User.Identity.IsAuthenticated AndAlso context.User.Identity.Name.ToUpper() = "CEO" Then
            DownloadFile(context)
        Else
            context.Response.Write("No access unless you're the CEO!!!")
        End If
    End Sub

    Protected Sub DownloadFile(ByVal context As HttpContext)
        context.Response.Buffer = True
        context.Response.Clear()
        context.Response.AddHeader("content-disposition", context.Request.Url.AbsolutePath)
        context.Response.ContentType = "text/plain"
        context.Response.WriteFile(context.Server.MapPath(context.Request.Url.AbsolutePath))
    End Sub

    Public Overrides Function ToString() As String
        Return "Some generic description to define what this class does.."
    End Function


    ReadOnly Property IsReusable() As Boolean Implements IHttpHandler.IsReusable
        Get
            Return False
        End Get
    End Property

End Class

File: CS/CS.NET/Visual Studio/IHttpHandler/IHttpHandler/HttpHandlerCS/CustomHandler.ashx.cs
using System;
using System.Web;

namespace WebApplication1
{   
    public class CustomHandler : IHttpHandler
    {
        public void ProcessRequest(HttpContext context)
        {
            if (context.User.Identity.IsAuthenticated &&
                context.User.Identity.Name.ToUpper() == "CEO")
            {
                DownloadFile(context);
            }
            else
            {
                context.Response.Write("No access unless you're the CEO!!!");
            }
        }

        protected void DownloadFile(HttpContext context)
        {
            context.Response.Buffer = true;
            context.Response.Clear();
            context.Response.AddHeader("content-disposition", context.Request.Url.AbsolutePath);
            context.Response.ContentType = "text/plain";
            context.Response.WriteFile(context.Server.MapPath(context.Request.Url.AbsolutePath));         
        }  

        public override string ToString()
        {
            return "Some generic description to define what this class does..";
        }

        public bool IsReusable
        {
            get
            {
                return false;
            }
        }
    }
}
